packages/inertia/src/router.ts

```diff
@@ -374,18 +376,24 @@ export class Router {
     replace = false,
     preserveScroll = false,
     preserveState = false,
+    preserveUrl = false,
   }: {
     visitId?: VisitId,
     replace?: boolean,
     preserveScroll?: PreserveStateOption
     preserveState?: PreserveStateOption
+    preserveUrl?: PreserveStateOption
   } = {}): Promise<void> {
     return Promise.resolve(this.resolveComponent(page.component)).then(component => {
       if (visitId === this.visitId) {
         page.scrollRegions = page.scrollRegions || []
         page.rememberedState = page.rememberedState || {}
         replace = replace || hrefToUrl(page.url).href === window.location.href
-        replace ? this.replaceState(page) : this.pushState(page)
+
+        if (! preserveUrl) {
+          replace ? this.replaceState(page) : this.pushState(page)
+        }
```

Comment:
This option should not be called `preserveUrl` but something like `saveState`. What it actually does is skip the addition of the current page data to the history, so it's not just skipping replacing the URL. 

If the wanted behavior is to keep the current URL, wouldn't it be better to do this?

```suggestion
        if (preserveUrl) {
        	page.url = window.location.href
        }
        
        replace ? this.replaceState(page) : this.pushState(page)
```